lib/lsp-data/api_pol_invoice_list.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module LspData
+  ### Represents the list of invoices returned from an invoice API call that
+  ###   finds invoices attached to a specific PO Line
+  class ApiPolInvoiceList
+    attr_reader :pol, :api_key, :conn
+
+    def initialize(pol:, api_key:, conn:)
+      @pol = pol
+      @api_key = api_key
+      @conn = conn
+    end
+
+    def invoices
+      @invoices ||= all_invoices
+    end
+
+    private
```

Comment:
I like the nice clean API of this class, with all of the implementation hidden in private methods!